credential-exchange-types/src/b64url.rs

```diff
@@ -0,0 +1,134 @@
+use data_encoding::{Specification, BASE32_NOPAD, BASE64URL, BASE64URL_NOPAD};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
+#[serde(try_from = "&str", into = "String")]
+pub struct B64Url(Vec<u8>);
+
+impl From<Vec<u8>> for B64Url {
+    fn from(src: Vec<u8>) -> Self {
+        Self(src)
+    }
+}
+impl From<&[u8]> for B64Url {
+    fn from(src: &[u8]) -> Self {
+        Self(src.to_vec())
+    }
+}
+
+impl From<B64Url> for Vec<u8> {
+    fn from(src: B64Url) -> Self {
+        src.0
+    }
+}
+
+impl AsRef<[u8]> for B64Url {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl From<B64Url> for String {
+    fn from(src: B64Url) -> Self {
+        String::from(&src)
+    }
+}
+impl From<&B64Url> for String {
+    fn from(src: &B64Url) -> Self {
+        BASE64URL_NOPAD.encode(&src.0)
+    }
+}
+
+impl std::fmt::Display for B64Url {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(String::from(self).as_str())
+    }
+}
+
+/// An error returned when a string is not base64 decodable.
+#[derive(Debug)]
+pub struct NotB64UrlEncoded;
```

Comment:
We use B64 naming here, but Base32 below. Should we use either full or abbrevated?